consultation_analyser/consultations/views/answers.py

```diff
@@ -376,6 +376,194 @@ def question_responses_json(
+    question = get_object_or_404(
+        models.Question.objects.select_related("consultation"),
+        slug=question_slug,
+        consultation__slug=consultation_slug,
+    )
+
+    # Parse filters from request
+    filters = parse_filters_from_request(request)
+
+    # Database-level aggregation using Django ORM hybrid approach
+    theme_aggregations = {}
+
+    if question.has_free_text:
+        # Build base query with all filters applied
+        response_filter = build_response_filter_query(filters, question)
+
+        # Get theme counts directly from database with JOIN
+        theme_counts = (
+            models.Theme.objects.filter(
+                responseannotation__response__in=models.Response.objects.filter(response_filter)
+            )
+            .values("id")
+            .annotate(count=Count("responseannotation__response"))
+            .order_by("id")
+        )
+
+        theme_aggregations = {str(theme["id"]): theme["count"] for theme in theme_counts}
+
+    return JsonResponse({"theme_aggregations": theme_aggregations})
+
+
+@user_can_see_dashboards
+@user_can_see_consultation
+def filtered_responses(
+    request: HttpRequest,
+    consultation_slug: str,
+    question_slug: str,
+):
+    """Standalone endpoint for getting paginated filtered responses"""
+    # Get the question object with consultation in one query
+    question = get_object_or_404(
+        models.Question.objects.select_related("consultation"),
+        slug=question_slug,
+        consultation__slug=consultation_slug,
+    )
+
+    # Parse filters from request
+    filters = parse_filters_from_request(request)
+
+    # Get filtered responses with themes (optimized with prefetching)
+    filtered_qs = get_filtered_responses_with_themes(question, filters)
+
+    # Pagination parameters
+    DEFAULT_PAGE_SIZE = 50
+    page_size = int(request.GET.get("page_size", DEFAULT_PAGE_SIZE))
+    page_num = int(request.GET.get("page", 1))
+
+    # Use Django's lazy pagination
+    paginator = Paginator(filtered_qs, page_size, allow_empty_first_page=True)
+    page_obj = paginator.page(page_num)
+
+    # Get total respondents count for this question (single query)
+    all_respondents_count = models.Response.objects.filter(question=question).count()
+
+    data = {
+        "all_respondents": [build_respondent_data(r) for r in page_obj.object_list],
+        "has_more_pages": page_obj.has_next(),
+        "respondents_total": all_respondents_count,
+        "filtered_total": paginator.count,
+    }
+
+    return JsonResponse(data)
+
+
+@user_can_see_dashboards
+@user_can_see_consultation
+def question_information(
+    request: HttpRequest,
+    consultation_slug: str,
+    question_slug: str,
+):
+    """Standalone endpoint for getting basic question information"""
+    # Get the question object with consultation in one query
+    question = get_object_or_404(
+        models.Question.objects.select_related("consultation"),
```

Comment:
select_related isnt dping any harm but nor is it doing any good here as its only for N+1 queries where here N=0